parser.py

```diff
@@ -31,10 +31,11 @@ def negated(self):
 
 
 class Implication(object):
-    def __init__(self, condition, constraint):
+    def __init__(self, condition, constraint, multi_condition=False):
```

Comment:
Given that such implications can not be produced by any code in the suite, are not covered by the specification and breaks assumptions made in other parts of the code, that's a big :-1:. This is the reference implementation that means to visualize and test the specification, and it's not going beyond that.

Reply:
This code from flatten3 is what triggered it for me:
```python
		elif isinstance(expr, AnyOfOperator):
			# || ( a b c ... ) -> [!b !c ...]? ( a )
			yield (conditions+[x.negated() for x in expr.constraint[1:]], expr.constraint[0])
```
If anyone wants to test solve.py with `|| ( a b c )` then they'll hit the assertion.

Reply:
Ok, you're right. I've removed the assert entirely then. Thanks.

Reply:
Cool!